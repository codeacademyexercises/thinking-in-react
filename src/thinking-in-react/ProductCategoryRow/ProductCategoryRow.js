import React, { Component } from 'react';
import ProductRow from '../ProductRow/ProductRow.js';
import '../App/App.css';

class ProductCategoryRow extends Component{
    render(){
        return (
            <div>
            <table>
                <tbody>
                <tr>
                    <th>{this.props.category}</th>
                </tr>
                <ProductRow category={this.props.category} items={this.props.product}/>
                </tbody>
            </table>
            </div>
        );
    };
};

export default ProductCategoryRow;