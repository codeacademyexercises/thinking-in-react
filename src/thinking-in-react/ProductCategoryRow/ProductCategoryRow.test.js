import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductCategoryRow from './ProductCategoryRow.js';

const store = [{category: 'Sporting Goods',Name: 'Football',Price: 49.99, stocked: true},
    {category: 'Sporting Goods',Name: 'Baseball',Price: 9.99, stocked: true},
    {category: 'Sporting Goods',Name: 'Basketball',Price: 29.99, stocked: false},
    {category: 'Electronics',Name: 'iPod Touch',Price: 99.99, stocked: true},
    {category: 'Electronics',Name: 'iPhone 5',Price: 399.99, stocked: false},
    {category: 'Electronics',Name: 'Nexus 7',Price: 199.99, stocked: true},
    ];

describe("<ProductCategoryRow /> should ",()=>{
    it("render",()=>{
        const tree = renderer.create(<ProductCategoryRow category="" product={store}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("have <ProductRow/> component and should assign props correctly",()=>{
        const wrapper = shallow(<ProductCategoryRow />);
        expect(wrapper.find('ProductRow')).toHaveLength(1);
        wrapper.setProps({
            category: ['Electronics'],
            product: [
                {category: 'Electronics',Name: 'iPod Touch',Price: 99.99, stocked: true},
            {category: 'Electronics',Name: 'iPhone 5',Price: 399.99, stocked: false},
            {category: 'Electronics',Name: 'Nexus 7',Price: 199.99, stocked: true},
            ],
        });
        expect(wrapper.find('ProductRow').props().category).toEqual(["Electronics"]);
        expect(wrapper.find('ProductRow').props().items).toEqual([store[3],store[4],store[5]]);
    });
});