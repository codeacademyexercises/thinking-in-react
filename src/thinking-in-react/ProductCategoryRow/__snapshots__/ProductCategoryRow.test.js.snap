// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCategoryRow /> should  render 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th>
          
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
